examples/controls_example/public/react_controls/types.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { BehaviorSubject } from 'rxjs';
+
+import { SerializedStyles } from '@emotion/react';
+import { ControlWidth } from '@kbn/controls-plugin/public/types';
+import { HasSerializableState } from '@kbn/presentation-containers';
+import { PanelCompatibleComponent } from '@kbn/presentation-panel-plugin/public/panel_component/types';
+import {
+  HasParentApi,
+  HasType,
+  HasUniqueId,
+  PublishesBlockingError,
+  PublishesDataLoading,
+  PublishesDisabledActionIds,
+  PublishesPanelTitle,
+  PublishesUnsavedChanges,
+  PublishingSubject,
+  StateComparators,
+} from '@kbn/presentation-publishing';
+
+import { ControlGroupApi } from './control_group/types';
+
+export interface PublishesControlDisplaySettings {
+  grow: PublishingSubject<boolean | undefined>;
+  width: PublishingSubject<ControlWidth | undefined>;
+}
+
+export interface HasCustomPrepend {
+  getCustomPrepend: () => React.FC<{}>;
+}
+
+/** This is the stuff the control group cares about */
+export type DefaultControlApi = PublishesDataLoading &
+  PublishesBlockingError &
+  PublishesUnsavedChanges &
+  PublishesControlDisplaySettings &
+  Partial<PublishesPanelTitle & PublishesDisabledActionIds & HasCustomPrepend> &
+  HasType &
+  HasUniqueId &
+  HasSerializableState &
+  HasParentApi<ControlGroupApi> & {
+    setDataLoading: (loading: boolean) => void;
+    setBlockingError: (error: Error | undefined) => void;
```

Comment:
I need to make setters for loading and error so that each control that uses either `initializeDefaultControlApi` or `initializeDataControl` can set these values - not sure if there is a better way to handle this?